### ISSUE TITLE
compiling for aarch64-unknown-linux-musl fails in binutils

### ISSUE THREAD

**Issue by tiolan** [crosstool-ng/crosstool-ng#2097]
Hi,

my goal is to build the `aarch64-unknown-linux-musl` toolchain statically linked.

So I started off with the default config of `arm-unknown-linux-musl` and adapted it via menuconfig:
- select "Bitness 64bit"
- select "Build Static Toolchain"
- un-select "Enable support for plugins"
This seems to be a requirement for statically compiled toolchains.

The build fails in the step "Install binutils for target" with:
```
[ALL  ]    /home/dev/x-tools/aarch64-unknown-linux-musl/lib/gcc/aarch64-unknown-linux-musl/13.2.0/../../../../aarch64-unknown-linux-musl/bin/ld: /home/dev/x-tools/aarch64-unknown-linux-musl/aarch64-unknown-linux-musl/sysroot/usr/lib/libiberty.a(obstack.o): relocation R_AARCH64_ADR_PREL_PG_HI21 against symbol `stderr' which may bind externally can not be used when making a shared object; recompile with -fPIC
[ALL  ]    /home/dev/x-tools/aarch64-unknown-linux-musl/lib/gcc/aarch64-unknown-linux-musl/13.2.0/../../../../aarch64-unknown-linux-musl/bin/ld: /home/dev/x-tools/aarch64-unknown-linux-musl/aarch64-unknown-linux-musl/sysroot/usr/lib/libiberty.a(obstack.o)(.text+0x4): unresolvable R_AARCH64_ADR_PREL_PG_HI21 relocation against symbol `stderr'
```
I tried adding -fPIC to various places, but no effect.
Also building shared instead static does not solve it.

I am aware that this toolchain makes use of experimental features.
But disabling them does not make any difference.

- crosstool-NG version
`1.26.0.43_4420873`
`built from latest master on 44208730a61f778b91f76d2feca76cf0a433c8c6`
- host OS and its version
`Linux ub22-dev 6.2.0-1018-azure #18~22.04.1-Ubuntu SMP Tue Nov 21 19:25:02 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux`
- `.config` is this:
[config_aarch64.txt](https://github.com/crosstool-ng/crosstool-ng/files/14110276/config_aarch64.txt)
- `build.log` is this:
[build.zip](https://github.com/crosstool-ng/crosstool-ng/files/14110282/build.zip)

Any idea what could be the root-cause?

**Comment by cpackham** [comment]
I don't know offhand why aarch64+musl won't work (it's certainly something that occurs in rust land (but that's llvm not gcc+binutils). I suspect it's just because no-one has attempted that configuration I'd gladly accept a patch to add a `aarch64-unknown-linux-musl` sample if someone gets it working.

**Comment by cpackham** [comment]
I gave it a quick go. The result is #2098

I think the biggest difference compared to your attempt is that `CT_BINUTILS_FOR_TARGET` is not set which is likely a hangover from the fact that you used `arm-unknown-linux-musleabi` as a starting point.

**Comment by bhundven** [comment]
@cpackham I'm also giving this a go. Also... (I'm a bit rusty) Isn't `eabi` default for arm?

**Comment by cpackham** [comment]
For `eabi` is not a thing for aarch64. I think you still need to say `eabi` for arm because of hysterical raisins.

**Comment by tiolan** [comment]
@cpackham 
I applied your sample config and compiling the toolchain worked fine (not tested the toolchain though).
If I understand correctly you switched off building binutils for the target. But what if any of the binutils libraries is needed? Shouldn't binutils also be compiled for the target?

**Comment by cpackham** [comment]
The absence of binutils hasn't been a problem for the aarch64-glibc target.

**Comment by cpackham** [comment]
Closed via #2098
